course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -99,14 +146,129 @@
 
                 courseInstance['sites'] = ci.sites;
             }
-            courseInstance['members'] = members.count;
 
             return courseInstance;
         };
 
-        dc.courseInstanceId = $routeParams.courseInstanceId;
+        dc.resetForm = function() {
+            dc.formDisplayData = angular.copy(dc.courseInstance);
+        };
 
-        dc.setCourseInstance($routeParams.courseInstanceId);
+        dc.resetFormFromUI = function() {
+            dc.resetForm();
+            dc.resetGlobalAlerts();
+            dc.alerts.global.formReset = {show:true};
+        };
 
+        dc.resetGlobalAlerts = function(scrollToAlerts) {
+            // reposition viewport at top of iframe so we can see
+            // notification messages
+            dc.alerts.global = {};  // reset any existing global alerts
+            // scroll to top of form by default (i.e. if optional scrollToAlerts
+            // argument is not passed in); can disable by setting scrollToAlerts
+            // to false
+            if (scrollToAlerts || scrollToAlerts == null) {
+                scrollTo(0, 0);
+            }
+        };
+
+        dc.submitCourseDetailsForm = function() {
+            dc.courseDetailsUpdateInProgress = true;
+            var postData = {};
+            var url = djangoUrl.reverse(dc.apiProxy,
+                [dc.apiBase + dc.courseInstanceId + '/']);
+            // we could also get these from editable properties on the DOM
+            var fields = [
+                'description',
+                'instructors_display',
+                'location',
+                'meeting_time',
+                'notes',
+                'short_title',
+                'sub_title',
+                'title'
+            ];
+            fields.forEach(function(field) {
+                postData[field] = dc.formDisplayData[field];
```

Comment:
Do we want to do this only if the field is dirty?  I guess I'm wondering if the semantics for this page should be "change the things I changed" or "make change everything to match what I see here".